### PR TITLE
Quick fix for atommethods to return empty residue group

### LINKED ISSUE #2879: Atomnames methods should handle empty group
## Expected behavior ##

`u.residues[[]]._get_prev_residues_by_resid()` returns an empty residue group. Same for

* _get_next_residues_by_resid
* psi_selections
* omega_selections
* chi1_selections

## Actual behavior ##

IndexError due to not checking length first.

```python
>>> u.residues[[]].omega_selections()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 862, in omega_selections
    nxtres = residues._get_next_residues_by_resid()
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 670, in _get_next_residues_by_resid
    u = residues[0].universe
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/groups.py", line 527, in __getitem__
    return self.level.singular(self.ix[item], self.universe)
IndexError: index 0 is out of bounds for axis 0 with size 0
```

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD,  GRO, PDB, TPR, XTC, TRR,  PRMncdf, NCDF

u = mda.Universe(PSF, DCD)
u.residues[[]].omega_selections()

```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) develop 2.0.0-dev
- Which version of Python (`python -V`)? 3.7
- Which operating system? MacOS

## Update ##
This has been partially fixed by #3089. The error message for methods such as ``omega_selections()`` is now:

```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF, DCD
>>> u = mda.Universe(PSF, DCD)
>>> u.residues[[]].omega_selections()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 1082, in omega_selections
    for r in nxt]
TypeError: 'int' object is not iterable
```

### PR DESCRIPTION
**Pull request by aditya-kamath**
Partially fixes #2879

Changes made in this Pull Request:
 - Returns empty residue group for _get_prev_residues_by_resid and _get_next_residues_by_resid

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
Thanks for this PR @aditya-kamath. Not a review (I'll probably let @lilyminium handle this as she has a much better understanding of the issue), but just some passing quick comments:

1) Please remember to add yourself to AUTHORS and add an entry to CHANGELOG
2) Since this is fixing a very specific failure, you should aim to add some tests for the edge case that this is attempting to fix.
3) In your PR message, could you add the issue number this is partially fixing so that reviewers know what this is trying to fix?
4) Please have a look & fix the PEP8speaks issue.

**Comment by aditya-kamath** [comment]
Okay, Thanks @IAlibay! Happy to put up my first PR. 
I assume by adding tests you mean that I write the code which reproduces this behavior. (?) 
I'll patch up the other tasks.

**Comment by IAlibay** [comment]
> I assume by adding tests you mean that I write the code which reproduces this behavior. (?)

Yes, we use `pytest` for code testing in MDAnalysis. Under `mdanalysis/testsuite/MDAnalysis/core` you'll find a file called `test_topologyattrs.py`. 

Within it you'll need to add one or more test functions that cover the edge case (i.e. ensuring that you get an empty residue group return) that is being fixed here. 

If you aren't used to pytest (please ignore this if that's not the case); I would suggest doing the tests for the two functions you have fixed now before moving on to fixing the rest. This way it's easier to work through the code with you. Apologies for the brief explanation of what's needed test-wise, I don't want to presume your current knowledge of `pytest`. Do let us know if you want a more detailed walkthrough of it all.

**Comment by IAlibay** [comment]
@aditya-kamath any reason why this was closed?

**Comment by aditya-kamath** [comment]
Hi, @IAlibay. I am a complete beginner with this. So, your explanation on pytest is helpful! :) 
I figure I should close this and reopen it when I have done the things you've asked for?

**Comment by lilyminium** [comment]
No need, @aditya-kamath, it's common to go back and forth on a pull request :-) Let's just leave it open for now -- it also allows tests to run as a check on code.

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py [hunk 1]
```diff
@@ -180,6 +180,11 @@ class TestAtomnames(TestAtomAttr):
     single_value = 'Ca2'
     attrclass = tpattrs.Atomnames
 
+class TestemptyAtomnames(TestAtomAttr):
```
Comment (lilyminium) [inline testsuite/MDAnalysisTests/core/test_topologyattrs.py:183]: A class seems a little overkill for this simple test, especially as it subclasses `TestAtomAttr` (and thereby runs all the tests defined there again). Could you just make this a function?

These should also be two separate tests, I think, as they are different methods.

File: package/AUTHORS [hunk 2]
```diff
@@ -151,7 +151,8 @@ Chronological list of authors
   - Nicholas Craven
   - Mieczyslaw Torchala
   - Haochuan Chen
-
+2021
+  - Aditya Kamath
```
Comment (lilyminium) [inline package/AUTHORS:155]: Could you just make sure there's an empty line after your name and before `External code`?

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py (hunk 1 above)
Comment (lilyminium) [reply testsuite/MDAnalysisTests/core/test_topologyattrs.py:183]: Also, the name of this class is a little misleading -- you're testing empty residue groups, and the contents of the atom names has fairly little to do with it. In fact, as the methods are transplanted to `ResidueGroup`, a more concise test could just look at what ``u.residues[[]]._get_next_residues_by_resid()`` returns.

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thanks for adding a test, @aditya-kamath! Just a few comments.

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py (hunk 1 above)
Comment (aditya-kamath) [reply testsuite/MDAnalysisTests/core/test_topologyattrs.py:183]: Hi, Thanks for the recommendations! I'll cut it down a bit!

**Comment by aditya-kamath** [comment]
Hi @IAlibay @lilyminium, it looks like the checks have passed! Thanks for your feedback and comments :D

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py (hunk 1 above)
Comment (aditya-kamath) [reply testsuite/MDAnalysisTests/core/test_topologyattrs.py]: Just wanted to let you know that I changed the class to functions in my latest commit. Hopefully, that is what you were looking for ?

**Comment by lilyminium** [comment]
It's looking good, thank you @aditya-kamath! Could you just address the pep8 comments in https://github.com/MDAnalysis/mdanalysis/pull/3089#issuecomment-753368103? Then you're done :-)

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py (hunk 1 above)
Comment (lilyminium) [reply testsuite/MDAnalysisTests/core/test_topologyattrs.py]: Thank you!

File: package/CHANGELOG [hunk 3]
```diff
@@ -21,6 +21,7 @@ The rules for this file:
   * 2.0.0
 
 Fixes
+  * atommethods(_get_prev/next_residues_by_resid) returns empty residue group when given empty residue group (Issue #3089)
```
Comment (lilyminium) [inline package/CHANGELOG:24]: Could you please also add your user name to the author list above!

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
Great, thank you @aditya-kamath! Congratulations on your first commit to MDAnalysis! 🎉

**Pull request merged**